sql_generators/glean_usage/glean_app_ping_views.py

```diff
@@ -66,26 +66,105 @@ def generate_per_app(self, project_id, app_info, output_dir=None):
         ):
             return
 
+        env = Environment(loader=FileSystemLoader(PATH / "templates"))
+        view_template = env.get_template("app_ping_view.view.sql")
+
         p = GleanPing(repo)
+        # generate views for all available pings
         for ping_name in p.get_pings():
+            release_only = False  # flag indicating that view only generated for release channel due to schema incompatibilites
+
             view_name = ping_name.replace("-", "_")
-            underlying_view_id = ".".join(
-                ["moz-fx-data-shared-prod", repo["app_id"].replace("-", "_"), view_name]
-            )
             full_view_id = f"moz-fx-data-shared-prod.{target_dataset}.{view_name}"
+
+            # generate a unioned schema that contains all fields of all ping tables across all channels
+            unioned_schema = Schema.empty()
+
+            # iterate through app_info to get all channels
+            for app in app_info:
+                app_dataset = app["document_namespace"].replace("-", "_")
+                schema = Schema.for_table(
+                    "moz-fx-data-shared-prod",
+                    app_dataset,
+                    view_name,
+                    partitioned_by="submission_timestamp",
+                )
+
+                try:
+                    unioned_schema.merge(schema, add_missing_fields=True)
+                except Exception as e:
+                    # if schema incompatibilities are detected, then only generate for release channel
+                    print(
+                        f"Cannot UNION 'moz-fx-data-shared-prod.{app_dataset}.{view_name}': {e}"
+                    )
+                    print("Generate view only for release channel.")
+
+                    unioned_schema = Schema.for_table(
+                        "moz-fx-data-shared-prod",
+                        release_app["document_namespace"].replace("-", "_"),
+                        view_name,
+                        partitioned_by="submission_timestamp",
+                    )
+                    release_only = True
+
```

Comment:
seems like this loop should break here:
```suggestion
                    break
```